dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java

```diff
@@ -736,7 +736,12 @@ public VersionTypeValidation validateWorkflowSet(Set<SourceFile> sourcefiles, St
                 validationMessage = "Primary descriptor is empty.";
             } else if (!content.contains("class: Workflow")) {
                 isValid = false;
-                validationMessage = "Requires class: Workflow.";
+                validationMessage = "A CWL workflow requires class: Workflow.";
+                if (content.contains("class: CommandLineTool") || content.contains("class: ExpressionTool")) {
+                    validationMessage += " This file contains class: ";
+                    validationMessage += (content.contains("class: CommandLineTool")) ? "CommandLineTool." : "ExpressionTool.";
+                    validationMessage += " Did you mean to register a tool?";
```

Comment:
Might be just me, but I think it'd look cleaner with String.format

Reply:
Another consideration is efficiency, this repeated concatenation is pretty inefficient. 
Consider a StringBuilder https://stackoverflow.com/questions/4645020/when-to-use-stringbuilder-in-java